op-node/rollup/finality/finalizer.go

```diff
@@ -236,33 +245,22 @@ func (fi *Finalizer) tryFinalize() {
 	// go through the latest inclusion data, and find the last L2 block that was derived from a finalized L1 block
 	for _, fd := range fi.finalityData {
 		if fd.L2Block.Number > finalizedL2.Number && fd.L1Block.Number <= fi.finalizedL1.Number {
-			// check if fd.L2Block.Number is finalized on Babylon
-			queryParams := &sdk.L2Block{
-				BlockHeight:    fd.L2Block.Number,
-				BlockHash:      fd.L2Block.Hash.String(),
-				BlockTimestamp: fd.L2Block.Time,
-			}
-			fi.log.Debug(
-				"babylon gadget query params",
-				"block_height", queryParams.BlockHeight,
-				"block_hash", queryParams.BlockHash,
-				"block_timestamp", queryParams.BlockTimestamp,
-			)
-			babylonFinalized, err := fi.babylonFinalityClient.QueryIsBlockBabylonFinalized(queryParams)
-			fi.log.Debug("babylon gadget query result", "babylon_finalized", babylonFinalized)
-
-			// If the error encountered is of type NoFpHasVotingPowerError, it should be ignored;
-			// for any other error types, emit a critical error event.
-			if err != nil && !errors.Is(err, sdk.ErrNoFpHasVotingPower) {
-				fi.emitter.Emit(rollup.CriticalErrorEvent{Err: fmt.Errorf("failed to check if block %d is finalized on Babylon: %w", fd.L2Block.Number, err)})
+			lastFinalizedBlock, err := fi.findFinalizedL2Block(fd.L2Block.Number, finalizedL2.Number)
+			if err != nil {
+				if lastFinalizedBlock != (eth.L2BlockRef{}) {
+					// set finalized block(s)
+					finalizedL2, finalizedDerivedFrom = fi.updateFinalized(lastFinalizedBlock, fd.L1Block)
```

Comment:
should we use `lastFinalizedBlock.L1Origin` here instead of `fd.L1Block`?

b/c `lastFinalizedBlock` might not be the same as `fd.L2Block.Number`

Reply:
this is probably another edge case that test coverage failed. we should have a test case for this

Reply:
fixed in https://github.com/babylonchain/optimism/pull/23/commits/38f662595b76067e505493fd412697e2b2ae4b94